feat: Migrate cron job from vexhub-crawler to vexhub repository

knqyf263 opened the pull request:
## Summary
Migrate the cron job workflow from vexhub-crawler repository to vexhub repository to prevent GitHub Actions from being disabled due to inactivity.

## Background
As discussed in https://github.com/aquasecurity/vexhub-crawler/discussions/40, the GitHub Actions workflow in vexhub-crawler was disabled due to repository inactivity. Since the vexhub repository has regular commits (timestamp updates in index.json), moving the workflow here should prevent future disabling due to inactivity.

## Changes
- Add `.github/workflows/cron.yaml` that runs daily to update VEX documents
- Build vexhub-crawler from source (pinned to commit [691c2aaad9e15493af3416edb96b787e954ef1f7](https://github.com/aquasecurity/vexhub-crawler/commit/691c2aaad9e15493af3416edb96b787e954ef1f7), current head)
- Use `secrets.GITHUB_TOKEN` instead of `ORG_REPO_TOKEN` for better security
- Apply GitHub Actions security best practices with explicit permissions

## Test Results
Successfully tested in fork: https://github.com/knqyf263/vexhub/actions/runs/15533094872

## Security Improvements
- Use minimal permissions (`contents: write`, `actions: read`)
- Replace `ORG_REPO_TOKEN` with built-in `GITHUB_TOKEN`
- Clone repositories to separate directories to prevent accidental commits

knqyf263 committed:
feat: Migrate cron job from vexhub-crawler to vexhub repository

- Add GitHub Actions workflow to run VEX document updates
- Build vexhub-crawler from source within the workflow
- Pin to vexhub-crawler commit 691c2aaad9e15493af3416edb96b787e954ef1f7

knqyf263 committed:
fix: Use go install to avoid binary path conflict

knqyf263 committed:
fix: Specify config file path for vexhub-crawler

DmitriyLewen commented on .github/workflows/cron.yaml:
```diff
@@ -0,0 +1,60 @@
+        with:
+          path: vexhub
+
+      - name: Check out vexhub-crawler
+        uses: actions/checkout@v4
+        with:
+          repository: ${{ github.repository_owner }}/vexhub-crawler
+          ref: ${{ env.VEXHUB_CRAWLER_COMMIT }}
+          path: vexhub-crawler
```
> Use secrets.GITHUB_TOKEN instead of ORG_REPO_TOKEN for better security

I thought `secrets.GITHUB_TOKEN` doesn't have permissions to checkout another repository.
Has this been changed?

knqyf263 replied:
Public repositories can be cloned even without a token.

knqyf263 replied:
https://github.com/knqyf263/vexhub/actions/runs/15533094872/job/43726110031

DmitriyLewen replied:
hm...
Then we can remove tokens in `vuln-list-update` (e.g. https://github.com/aquasecurity/vuln-list-update/blob/e2493057791fda5d182e9d5e14b3c14e02e0384e/.github/workflows/nvd.yml#L27) 😄

knqyf263 replied:
We need it because the workflow pushes the commits, which requires write permission, not just read permission.

knqyf263 replied:
If a workflow pushes to another repository, it needs a token with write permissions. If it only needs read access, a token is not required, but for private repositories, a token is necessary. This is my understanding.

DmitriyLewen replied:
Got it. Thanks for the explanation!

knqyf263 replied:
This time, we are simply cloning and building the `vexhub-crawler`, so there’s no need to modify anything in the `vexhub-crawler` itself — read permissions are sufficient. From a security perspective, it's safer to use `secrets.GITHUB_TOKEN` in the data repository and clone the script from another repository.

The reason I didn’t do this in `vuln-list` is because `vuln-list` has a high commit frequency and a large amount of data. Pulling the entire `vuln-list` just to slightly modify a workflow would be time-consuming and inefficient, so I chose to add the workflow on the `vuln-list-update` side and use `ORG_REPO_TOKEN` instead.

This approach has also brought additional benefits over time. Specifically, we can now manage workflows such as those for `vuln-list-redhat` and `vuln-list-debian` in a single location. However, this comes with a trade-off in terms of security (since it requires `ORG_REPO_TOKEN`), so we might consider changing it in the future.

I just wanted to share my current thinking on this.

DmitriyLewen commented on .github/workflows/cron.yaml:
```diff
@@ -0,0 +1,60 @@
+    name: Update VEX Hub
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
I don't see any use for this env.
Can you explain why we need this?

knqyf263 replied:
I just copied your workflow 😄 Do you remember why you added it?
https://github.com/aquasecurity/vexhub-crawler/pull/20

DmitriyLewen replied:
oops 😄 
Perhaps i wanted to use `env.GITHUB_TOKEN`, but still used `secrets.ORG_REPO_TOKEN` in `actions/checkout`.

So we can remove this 😄

knqyf263 replied:
Removed https://github.com/aquasecurity/vexhub/pull/6/commits/1f889dec7cf9a390d8e93c8996df343652c79eba

knqyf263 committed:
fix: Remove unnecessary GITHUB_TOKEN from cron workflow

The GITHUB_TOKEN is automatically provided by GitHub Actions and doesn't need to be explicitly set in the environment variables.

DmitriyLewen commented on .github/workflows/cron.yaml:
```diff
@@ -0,0 +1,59 @@
+jobs:
+  update:
+    name: Update VEX Hub
+    runs-on: ubuntu-latest
```
nit:
I understand that we don't use any third party utilities (except git), but I think it's better to pin the Ubuntu version to avoid unexpected errors after upgrading the Ubuntu version in Runner (like we got the error with oras in trivy-db and trivy-java-db)

knqyf263 replied:
done https://github.com/aquasecurity/vexhub/pull/6/commits/5dc241c2bf600bbb65385dc046f74799634f8ff3

knqyf263 committed:
fix: Pin Ubuntu version to 24.04 in GitHub Actions workflow

knqyf263 commented:
It works.
https://github.com/aquasecurity/vexhub/actions/runs/15552217353/job/43784857685
